journal_brief/journal_brief.py

```diff
@@ -153,10 +153,24 @@ def __init__(self, cursor_file=None, reader=None, dry_run=False,
         if reader is None:
             reader = journal.Reader()
 
-        if seek_cursor and self.cursor:
-            log.debug("Seeking to %s", self.cursor)
-            reader.seek_cursor(self.cursor)
-            reader.get_next()
+        if self.cursor:
+            if seek_cursor:
+                log.debug("Seeking to %s", self.cursor)
+                reader.seek_cursor(self.cursor)
+                reader.get_next()
+        else if not dry_run:
+            # use an unfiltered Reader to find the current 'tail'
+            # of the journal and store that as the initial cursor
+            # when the cursor file could not be found; this avoids
+            # reading through the entire journal again on the next
+            # run if the inclusions and exclusions result in zero
+            # matching entries during this run
+            temp_reader = journal.Reader()
```

Comment:
Instead of a `temp_reader`, we could initialise `reader` before trying to open the cursor file, and `seek_head()` afterwards. It's faster.

Reply:
I originally considered that, but `reader` can be passed as an argument to this constructor and it already has inclusions and exclusions in it (if any have been defined), in which case the seek/get_previous operation won't find any matching journal entries, and we'll end up with an empty cursor again.

Reply:
Aha, I did not remember that.